### DISCUSSION TITLE
const keyword in arguments

### DISCUSSION THREAD

**Discussion by YoussefAminGomaa** [dotnet/roslyn#67564] [Ideas] [answered]
add a const keyword to arguments like the Attribute the argument is not constant, the value is the constant. here's an example of it:

// the user should enter a constant value
WriteConstant("Hello, world!");
// the compiler should give an error if the user entered something like this:
string text = "Hello, world!";
WriteConstant(text);
// however, if the variable is constant then it's fine.

static void WriteConstant(const string text)
{
    Console.WriteLine(text);
}

**Comment by CyrusNajmabadi** [comment, marked as answer]
You could do this by adding an attribute to the parameter, then writing an analyzer that checks callsites for you. :)

**Reply by YoussefAminGomaa** [reply]
I didn't understand, could you please explain more or show an example of it?

**Reply by CyrusNajmabadi** [reply]
```c#
static void WriteConstant([Const] string text)
{
     Console.WriteLine(text);
}
```

Then write an analyzer that checks callers of `WriteConstant` and errors if you pass it something non-constant.

**Reply by YoussefAminGomaa** [reply]
Thanks

**Reply by hez2010** [reply]
You don't have to write an analyzer, we already have `ConstantExpected` for this exact use case. 
See https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.constantexpectedattribute

**Reply by YoussefAminGomaa** [reply]
Too late, I actually wrote my own analyzer and it worked.

**Reply by Xyncgas** [reply]
In C++ you can just say static void WriteConstant(const string text)
Is there any reason this is not in C# aside from no one bothered to implement it
